### PR TITLE
Added setuptools entry point for Sphinx

### PR DESCRIPTION
**Pull request by segevfiner**
This allows using `sphinx_rtd_theme` as Sphinx `html_theme` directly.

### CODE REVIEW

**Comment by ericholscher** [comment]
Thanks. Is this the same code that was used in Sphinx previously? It would be good to have a comment or something that points to an explanation of how this works.

**Comment by segevfiner** [comment]
I added a link to the section in Sphinx's documentation about this. In essence, this is just a classic usage of the setuptools/pkg_resources entry points mechanism. Documented [here](https://setuptools.readthedocs.io/en/latest/setuptools.html#dynamic-discovery-of-services-and-plugins) and [here](https://setuptools.readthedocs.io/en/latest/pkg_resources.html#entry-points). But there might be more explanations elsewhere too 😛

File: README.rst [hunk 1]
```diff
@@ -48,6 +48,12 @@ In your ``conf.py`` file:
 
     html_theme_path = [sphinx_rtd_theme.get_html_theme_path()]
 
+or (since master):
```
Comment (Blendify) [inline README.rst]: I think this text should be handled differently maybe use `.. versionchanged::` or `.. versionadded::` referencing the next major release of the theme (0.2.5)

Comment (segevfiner) [inline README.rst]: I think GitHub's reStructuredText renderer doesn't know about `.. versionadded::`/`.. versionchanged::` which are Sphinx things.

Comment (Blendify) [inline README.rst]: I would just say as of 0.2.5... then

**Comment by ericholscher** [comment]
Thanks!

**Pull request merged**